validator/src/journal/publisher.rs

```diff
@@ -589,35 +516,27 @@ impl BlockPublisher {
             data_dir,
             config_dir,
             permission_verifier,
-            check_publish_block_frequency,
             batch_observers,
             batch_injector_factory,
-            consensus_factory,
             block_wrapper_class,
             block_header_class,
             block_builder_class,
             settings_view_class,
             exit: Arc::new(Exit::new()),
         };
 
-        BlockPublisher {
-            publisher,
-            batch_tx,
-            batch_rx: Some(batch_rx),
-        }
+        BlockPublisher { publisher }
     }
 
-    pub fn start(&mut self) {
+    pub fn start(&mut self) -> IncomingBatchSender {
+        let (batch_tx, mut batch_rx) = make_batch_queue();
         let builder = thread::Builder::new().name("PublisherThread".into());
-        let mut batch_rx = self.batch_rx.take().unwrap();
-        let mut block_publisher = self.publisher.clone();
+        let block_publisher = self.publisher.clone();
         builder
             .spawn(move || {
-                let mut now = Instant::now();
-                let check_period = { block_publisher.check_publish_block_frequency };
                 loop {
                     // Receive and process a batch
-                    match batch_rx.get(check_period) {
+                    match batch_rx.get(Duration::from_millis(100)) {
```

Comment:
What's the consequence of this 100 millisecond timeout?

Reply:
The timeout allows the loop to check to see if the process has been flagged to exit.  

A possibly better solution would be to use exit channels that we can select on, but this requires a discussion about channel libraries, since this is not yet supported in the standard library.